webapp/src/js/controllers/inbox.js

```diff
@@ -421,31 +436,32 @@ var _ = require('underscore'),
               $rootScope.$broadcast('formLoadingComplete');
             }
           );
+          // get the forms for the Add Report menu
+          XmlForms('AddReportMenu', { contactForms: false }, function(err, xForms) {
```

Comment:
Should this return a promise so things load in series instead of in parallel?

Reply:
The XmlForms service listens for changes and calls the callback if a form is added, removed, or modified. We're not allowed to resolve promises multiple times so a callback is required.

A cleaner pattern would be the listener pattern but that's just syntactic sugar for the same runtime experience.

The ideal pattern would have a promise based initial request, and also register a listener for subsequent changes but that's a more significant change...